app/src/androidTest/java/ch/epfl/polybazaar/FillListingActivityTest.java

```diff
@@ -349,16 +336,40 @@ public void cancelTakingPicture() throws Throwable {
         Intents.init();
         expectedCameraIntent = hasAction(MediaStore.ACTION_IMAGE_CAPTURE);
         intending(expectedCameraIntent).respondWith(cameraResult);
-
-        runOnUiThread(() -> {
-            Button but = fillSaleActivityTestRule.getActivity().findViewById(R.id.camera);
-            but.performClick();
-        });
+        runOnUiThread(() -> fillSaleActivityTestRule.getActivity().findViewById(R.id.addImage).performClick());
+        ViewInteraction appCompatButton2 = onView(
+                Matchers.allOf(withId(android.R.id.button1), withText(R.string.camera),
```

Comment:
I know this is not part of your PR, but renaming this to an explicit name could also help the comprehesion.